src/filters/std/string/slice.rs

```diff
@@ -0,0 +1,402 @@
+use filters::invalid_argument;
+use itertools;
+use liquid_compiler::{Filter, FilterParameters};
+use liquid_derive::*;
+use liquid_error::Result;
+use liquid_interpreter::Context;
+use liquid_interpreter::Expression;
+use liquid_value::Value;
+use std::cmp;
+use unicode_segmentation::UnicodeSegmentation;
+
+fn canonicalize_slice(
+    slice_offset: isize,
+    slice_length: isize,
+    vec_length: usize,
+) -> (usize, usize) {
+    let vec_length = vec_length as isize;
+
+    // Cap slice_offset
+    let slice_offset = cmp::min(slice_offset, vec_length);
+    // Reverse indexing
+    let slice_offset = if slice_offset < 0 {
+        slice_offset + vec_length
+    } else {
+        slice_offset
+    };
+
+    // Cap slice_length
+    let slice_length = if slice_offset + slice_length > vec_length {
+        vec_length - slice_offset
+    } else {
+        slice_length
+    };
+
+    (slice_offset as usize, slice_length as usize)
+}
+
+#[derive(Debug, FilterParameters)]
+struct SliceArgs {
+    #[parameter(description = "The offset of the slice.", arg_type = "integer")]
+    offset: Expression,
+
+    #[parameter(description = "The length of the slice.", arg_type = "integer")]
+    length: Option<Expression>,
+}
+
+#[derive(Clone, ParseFilter, FilterReflection)]
+#[filter(
+    name = "slice",
+    description = "Takes a slice of a given string or array.",
+    parameters(SliceArgs),
+    parsed(SliceFilter)
+)]
+pub struct Slice;
+
+#[derive(Debug, FromFilterParameters, Display_filter)]
+#[name = "slice"]
+struct SliceFilter {
+    #[parameters]
+    args: SliceArgs,
+}
+
+impl Filter for SliceFilter {
+    fn evaluate(&self, input: &Value, context: &Context) -> Result<Value> {
+        let args = self.args.evaluate(context)?;
+
+        let offset = args.offset as isize;
+        let length = args.length.unwrap_or(1) as isize;
+
+        if length < 1 {
+            return invalid_argument("length", "Positive number expected").into_err();
+        }
+
+        if let Value::Array(input) = input {
```

Comment:
The downside of categorizing this as `string` is it also supprts `array`.

It is really hard to figure out how to slice/dice the filters :)

Reply:
My bad, didn't notice that!

Reply:
This made me notice, by the way, that there aren't any unit tests for slice, even in the original filters implmentation.